docs/proposals/cloud-edge-custom-message-design.md

```diff
@@ -0,0 +1,429 @@
+                    rule-endpoint type its value is {"path":"/a/b"}. For eventbus rule-endpoint type its
+                    value is {"topic":"<user define string>","node_name":"xxxx"}'
+                  type: object
+                  additionalProperties:
+                    type: string
+                target:
+                  description: 'target is a string value representing where the messages go to. its value is
+                     the same with rule-endpoint name. For example, my-eventbus or my-rest or my-servicebus.'
+                  type: string
+                targetResource:
+                  description: 'targetResource is a map representing the resource info of target. For rest
+                    rule-endpoint type its value is {"resource":"http://a.com"}. For eventbus rule-endpoint
+                    type its value is {"topic":"/xxxx"}. For servicebus rule-endpoint type its value is {"path":"/request_path"}.'
+                  type: object
+                  additionalProperties:
+                    type: string
+              required:
+                - source
+                - sourceResource
+                - target
+                - targetResource
+            status:
+              type: object
+              properties:
+                successMessages:
+                  type: integer
+                failMessages:
+                  type: integer
+                errors:
+                  items:
+                    type: string
+                  type: array
+              required:
+                - successMessages
+                - failMessages
+                - errors
+  scope: Namespaced
+  names:
+    plural: rules
+    singular: rule
+    kind: Rule
+    shortNames:
+      - rule
+
+```
+
+### how to deliver custom messages
+
+1. cloud to edge : **rest->eventbus**
+
+1.1 create rest and eventbus type ruleEndpoint if they don't exist. Exec command:
+
+	`kubectl create -f create-ruleEndpoint-rest.yaml;kubectl create -f create-ruleEndpoint-eventbus.yaml`
+
+
+create-ruleEndpoint-rest.yaml's contents are as follows:
+```
+apiVersion: rules.kubeedge.io/v1
+kind: RuleEndpoint
+metadata:
+  name: my-rest
+  labels:
+    description: test
+spec:
+  ruleEndpointType: "rest"
+  properties: {}
+```
+
+create-ruleEndpoint-eventbus.yaml's contents are as follows:
+```
+apiVersion: rules.kubeedge.io/v1
+kind: RuleEndpoint
+metadata:
+  name: my-eventbus
+  labels:
+    description: test
+spec:
+  ruleEndpointType: "eventbus"
+  properties: {}
+```
+
+
+1.2 create rule. Exec command:
+
+`kubectl create -f create-rule-rest-eventbus.yaml`
+
+create-rule-rest-eventbus.yaml's contents are as follows:
+
+```
+apiVersion: rules.kubeedge.io/v1
+kind: Rule
+metadata:
+  name: my-rule
+  labels:
+    description: test
+spec:
+  source: "my-rest"
+  sourceResource: {"path":"/a"}
+  target: "my-eventbus"
+  targetResource: {"topic":"test"}
+status:
```

Comment:
status field not needed when call creation operation.

Reply:
status is used for custom field to record successful and failed message count.